MapboxDirections/MBDirections.swift

```diff
@@ -141,10 +141,17 @@ open class Directions: NSObject {
         
         self.accessToken = accessToken!
         
+        if(host == nil || host!.isEmpty){
```

Comment:
A cleaner way to write this line would be:

```swift
if host?.isEmpty ?? true {
```

But let’s strive for consistency with how we handle access tokens in Info.plist. Up at the top of the file, define a `defaultHost` constant that lazily reads the value out of Info.plist, like this:

https://github.com/mapbox/MapboxDirections.swift/blob/696694cf5f7448415e2009e32e7286b1ea6b630c/MapboxDirections/MBDirections.swift#L6-L7

Then, define a local variable `host` that shadows the initializer’s argument, falling back to `defaultHost`, finally falling back to `api.mapbox.com`:

https://github.com/mapbox/MapboxDirections.swift/blob/696694cf5f7448415e2009e32e7286b1ea6b630c/MapboxDirections/MBDirections.swift#L139

(The assertion isn’t needed because there will always be a fallback to `api.mapbox.com` in the worst-case scenario.)